### PR TITLE
Scheduler ci yaml merge

### PR DESCRIPTION
**Pull request by jtmcdole**
CiYaml now tracks one or more ci.yaml files in a fusion tree

CiYamlInner is the old CiYaml. Instead of merging or updating protobufs to know which target are engine targets and which are framework targets; keep them apart. We'll have to use this information later when scheduling engine builds first and waiting before we can schedule the test runs.

The fusion-engine-ciyaml will need to be updated to include paths; that needs to be added to the merge script.

- Downloading a ci.yaml path should respect the path requested
- Only adding the ability to find targets; scheduling to follow this PR.

The `settings.json` file has only the dart line length specified; which should be a shared setting between all developers as it is enforced by CI.

### CODE REVIEW

File: app_dart/lib/src/model/ci_yaml/ci_yaml.dart [hunk 1]
```diff
@@ -11,19 +11,130 @@ import 'package:github/github.dart';
...
+  /// Get's the [CiYamlInner] for the requested [type].
+  ///
+  /// The type is expected to exist and will fail otherwise.
+  CiYamlInner innerFor(CiType type) => configs[type]!;
+
+  /// The [RepositorySlug] that [config] is from.
+  final RepositorySlug slug;
+
+  /// The git branch currently being scheduled against.
+  final String branch;
```
Comment (yjbanov) [inline app_dart/lib/src/model/ci_yaml/ci_yaml.dart:69]: Feels wrong for something that represents the ci.yaml config to be "scheduled against" some branch. Can we separate the scheduling concern from the configuration?

File: app_dart/lib/src/model/ci_yaml/ci_yaml.dart [hunk 2]
```diff
@@ -11,19 +11,130 @@ import 'package:github/github.dart';
...
+  }) =>
+      configs[type]!.getFirstPostsubmitTarget(builderName);
+
+  /// List of target names used to filter target from release candidate branches
+  /// that were already removed from main.
+  List<String>? totTargetNames({CiType type = CiType.any}) => configs[type]!.totTargetNames;
+
+  /// List of postsubmit target names used to filter target from release candidate branches
+  /// that were already removed from main.
+
```
Comment (yjbanov) [inline app_dart/lib/src/model/ci_yaml/ci_yaml.dart:90]: nit: unnecessary empty line

File: app_dart/lib/src/service/scheduler.dart [hunk 3]
```diff
@@ -674,6 +699,8 @@ class Scheduler {
   /// Relevant APIs:
   ///   https://developer.github.com/v3/checks/runs/#check-runs-and-requested-actions
   Future<bool> processCheckRun(cocoon_checks.CheckRunEvent checkRunEvent) async {
+    /// Figure out if we're in fusion or not.
```
Comment (yjbanov) [inline app_dart/lib/src/service/scheduler.dart:702]: Should this be a TODO? Also, nit: probably not `///`

**Review by yjbanov — APPROVED** [review]
Just nitpicks. LGTM.

File: app_dart/lib/src/model/ci_yaml/ci_yaml.dart (hunk 1 above)
Comment (jtmcdole) [reply app_dart/lib/src/model/ci_yaml/ci_yaml.dart:69]: This is copy-pasta from the old CiYaml and I don't want to make big logical changes at the moment. My understanding is this is specifically for release branch handling - and it looks inside the ci yaml for special bits.

File: app_dart/lib/src/model/ci_yaml/ci_yaml.dart (hunk 2 above)
Comment (jtmcdole) [reply app_dart/lib/src/model/ci_yaml/ci_yaml.dart]: done

File: app_dart/lib/src/service/scheduler.dart (hunk 3 above)
Comment (jtmcdole) [reply app_dart/lib/src/service/scheduler.dart]: done

**Review by jtmcdole — COMMENTED** [review]
done

File: app_dart/lib/src/foundation/utils.dart [hunk 4]
```diff
@@ -22,6 +22,7 @@ import '../request_handling/exceptions.dart';
 import '../service/logging.dart';
 
 const String kCiYamlPath = '.ci.yaml';
+const String kCiYamlFusionPath = 'engine/src/flutter/$kCiYamlPath';
```
Comment (christopherfujino) [inline app_dart/lib/src/foundation/utils.dart:25]: should this be `kCiYamlFusionEnginePath`?

File: app_dart/lib/src/model/ci_yaml/ci_yaml.dart [hunk 5]
```diff
@@ -11,19 +11,129 @@ import 'package:github/github.dart';
...
+  /// Post-fusion: framework, etc
+  ///
+  /// NOTE: Required to be present
+  any,
+
+  /// Engine's .ci.yaml file located in the monorepo
+  fusionEngine,
+}
+
+/// Wrapper arround one or more [CiYaml]
```
Comment (christopherfujino) [inline app_dart/lib/src/model/ci_yaml/ci_yaml.dart:28]: Can you clarify more? Why do we need an abstraction around multiple CiYamls?

Also, should maybe CiYamlInner be named CiYaml, and this be named something else, maybe CiYamlSet?

File: app_dart/lib/src/model/proto/internal/scheduler.proto [hunk 6]
```diff
@@ -27,7 +27,7 @@ message SchedulerConfig {
 }
 
 // A unit of work for infrastructure to run.
-// Next ID: 17
+// Next ID: 18
```
Comment (christopherfujino) [inline app_dart/lib/src/model/proto/internal/scheduler.proto:30]: Should this not be 19? Also, why do we need this?

**Review by christopherfujino — APPROVED** [review]
LGTM

File: app_dart/lib/src/model/ci_yaml/ci_yaml.dart (hunk 5 above)
Comment (jtmcdole) [reply app_dart/lib/src/model/ci_yaml/ci_yaml.dart:28]: > Can you clarify more

Yes, I'll take a stab

> CiYaml vs CiYamlSet

I originally wanted a smaller diff-set and file changes.  If I change it to CiYamlSet, would folks expect it to actually be a Set<CiYaml>?  I'll give it a stab and see how it looks (at the call site, I'm sure it'll read better).

File: app_dart/lib/src/model/proto/internal/scheduler.proto (hunk 6 above)
Comment (jtmcdole) [reply app_dart/lib/src/model/proto/internal/scheduler.proto:30]: Ah, it should be. This looks like a relic of ye'olden times.  I'll just back this out and we can have a single PR updating this.

**Pull request merged**